### ISSUE TITLE
github_branch_protection does not enable "Require status checks to pass before merging"

### ISSUE THREAD

**Issue by gertjanmaas** [integrations/terraform-provider-github#880]
### Terraform Version
```
➜ terraform --version
Terraform v1.0.3
on darwin_amd64
+ provider registry.terraform.io/hashicorp/github v4.13.0
+ provider registry.terraform.io/integrations/github v4.13.0
```

### Affected Resource(s)
- `github_branch_protection`

### Terraform Configuration Files
```hcl
resource "github_branch_protection" "main_branch" {
  repository_id  = github_repository.my_test_repo.name

  pattern                 = "main"
  enforce_admins   = true

  required_status_checks {
    strict  = true
  }

  required_pull_request_reviews {
    required_approving_review_count = 1
  }
}
```

### Expected Behavior
"Require status checks to pass before merging" and "Require branches to be up to date before merging" options in branch protection should be enabled.

### Actual Behavior
"Require status checks to pass before merging" is disabled and "Require branches to be up to date before merging" is enabled.

### Steps to Reproduce
1. Use the configuration above
2. `terraform apply`

### Important Factoids
We upgraded from github provider `3.0.0` to `4.13.0` a while ago and recently noticed that this configuration behaves differently. 
In 3.0.0 setting `strict = true` in `required_status_checks` would enable both "Require status checks to pass before merging" and "Require branches to be up to date before merging".
In 4.13.0 it only enables "Require branches to be up to date before merging", which is weird because that option in the UI is hidden behind "Require status checks to pass before merging".

Note that the intention here is to require all status checks to pass before merging. Passing an empty `contexts = []` in the `required_status_checks` block does not seem to help and we're applying these branch protection rules on multiple repositories where we don't know the context names.

**Comment by ShaunEgan** [comment]
Can confirm, I am experiencing the same issue with the `4.13.0` version.

**Comment by martinmigasiewicz-tomtom** [comment]
I am facing this problem as well on 4.14 and realized that it works if I go back to `github_branch_protection_v3`.

**Comment by kfcampbell** [comment]
I've confirmed this behavior. Using the above HCL produces a weird state in the GitHub UI where status checks aren't required before merging:
![image](https://user-images.githubusercontent.com/9327688/136291394-0b101d19-de47-411f-b7d6-654de453c1ab.png)

But upon clicking that checkbox, the "up to date" checkbox appears, already checked: 
![image](https://user-images.githubusercontent.com/9327688/136291446-d701538c-7746-4771-9fe1-3ad76d7586b3.png)

which is not the usual behavior of the checkbox when purely operating through the GitHub API.

Upon further debugging, the issue happens because `ok` is not returning true in [this line of code](https://github.com/integrations/terraform-provider-github/blob/main/github/util_v4_branch_protection.go#L157). I am still unsure of what's causing it to be set that way. I've determined that it's _not_ related to an `Update` call, so something weird is happening on the initial creation. 

This definitely requires more investigation.
